src/pages/Cohort/Cohort.tsx

```diff
@@ -1,69 +1,141 @@
-import React, { useState } from "react";
-import UserTableHeader from "../../components/ui/UserTableHeader";
-import {
-  cohortTableDataItems,
-  cohortTableHeaders,
-  cohortTableSortingValues,
-  cohortsPerPage,
-} from "../../utils/data";
-import Button from "../../components/ui/Button";
-import PlusIcon from "../../assets/PlusIcon";
-import { getCohorts, getJWT } from "../../utils/helper";
-import AddingCohortModal from "../../components/modals/AddingCohort";
-import UserTable from "../../components/ui/UserTable";
-import { useGetAllCohortsQuery } from "../../features/user/apiSlice";
+import React from 'react'
+import { getJWT } from '../../utils/helper'
+import { useGetAllCohortsQuery } from '../../features/user/apiSlice'
+import { DataGrid, GridColDef } from '@mui/x-data-grid'
+import { ButtonSize } from '../../utils/types'
+import Button from '../../components/ui/Button'
+import CohortSkeleton from './CohortSkeleton'
+import ActionButtons from './ActionButtons'
+import SearchBar from './SearchBar'
 
-const Cohort = () => {
-  const jwt: string = getJWT();
-  const [query, setQuery] = useState("");
-  const [isAddingCohort, setIsAddingCohort] = useState(false);
+type TCohort = {
+  applicants: number
+  coaches: number
+  description: string
+  forms: number
+  name: string
+  stages: number
+  trainees: number
+  _id: string
+}
+type TCohortWithId = TCohort & { readonly id: number }
+const columns: GridColDef<TCohortWithId>[] = [
+  {
+    field: 'id',
+    headerName: 'No.',
+    sortable: false,
+    align: 'center',
+    headerAlign: 'center',
+    headerClassName: 'header-background',
+  },
+  {
+    field: 'name',
+    headerName: 'Name',
+    flex: 1,
+    sortable: false,
+    align: 'center',
+    headerAlign: 'center',
+    headerClassName: 'header-background',
+  },
+  {
+    field: 'stages',
+    headerName: 'Stages',
+    flex: 1,
+    sortable: false,
+    align: 'center',
+    headerAlign: 'center',
+    headerClassName: 'header-background',
+  },
+  {
+    field: 'trainees',
+    headerName: 'Current participants',
+    type: 'number',
+    flex: 1,
+    sortable: false,
+    align: 'center',
+    headerAlign: 'center',
+    headerClassName: 'header-background',
+  },
+  {
+    field: 'coaches',
+    headerName: 'Coaches',
+    flex: 1,
+    sortable: false,
+    align: 'center',
+    headerAlign: 'center',
+    headerClassName: 'header-background',
+  },
+  {
+    field: 'actions',
+    headerName: 'Actions',
+    flex: 1,
+    sortable: false,
+    align: 'center',
+    headerAlign: 'center',
+    headerClassName: 'header-background',
+    renderCell: () => {
+      return <ActionButtons />
+    },
+  },
+]
```

Comment:
Impressive! Is there a way we can organize the columns array into a separate file? it's making the component quite large!